ygen/genir_test.go

```diff
@@ -141,11 +171,17 @@ func TestGenerateIR(t *testing.T) {
 						"one": {
 							Name: "One",
 							YANGDetails: YANGNodeDetails{
-								Name:         "one",
-								Defaults:     nil,
-								Module:       "openconfig-simple",
-								Path:         "/openconfig-simple/parent/child/config/one",
-								ResolvedPath: "",
+								Name:            "one",
+								Defaults:        nil,
+								BelongingModule: "openconfig-simple",
+								RootModule:      "openconfig-simple",
+								DefiningModule:  "openconfig-simple",
+								Path:            "/openconfig-simple/parent/child/config/one",
+								SchemaPath:      "/parent/child/config/one",
+								ResolvedPath:    "",
+								Description:     "",
+								Kind:            "leaf",
+								Type:            &YANGType{Name: "string"},
```

Comment:
The use case is that in OpenConfig we use `ieeefloat32`, which is defined as a `binary` type. This is not user-friendly to Ondatra users, so in its telemetry API (currently being developed afresh in ygnmi) we have a special case for it where we check for this and convert the binary to a float32 before sending back the data to the user, and vice versa. If we don't have this information in the IR it would not be possible to 1) keep the ygot definition a binary due to the schema requirements, as we still need to marshal these as binary fields, and 2) allow the later passes to treat these binary fields differently from other binary fields.

I also don't like this, because if this is the way we solve these kinds of problems the IR will bloat. I'm thinking some sort of plugin mechanism should exist in the IR -- if we abstract things away we're hiding things, and that means some information is not accessible which could cause problems for certain use cases. To avoid making the IR as complicated as the layer underneath, then we need an extension mechanism. What do you think?

Reply:
Makes sense :-)

That sounds great to me -- why don't we have something like `Flags map[string]string` within the IR. Flags would be something that could be populated by a `SetFlag` in the `LangMapper`. This way, you could have the mapper be handed some specific thing that says "match on this type name and set this flag" and then the output can handle that without leaking the information post-IR phase?

Reply:
I think this is a good idea. I was considering whether if `ExtraInfo map[string]interface{}` is more flexible since it allows a user-defined type to be inserted. However, if we aim to keep the IR generally applicable to code generation, then it should have enough information such that the later passes should not need complex data being provided to it, so as you suggested using just `string` to indicate this. I'll make this a clean-up task and let's try how well it works in practice.

Reply:
That sounds reasonable to me (`map[string]interface{}`). Shall we file issues for these extension tasks?